clarity/src/vm/database/clarity_db.rs

```diff
@@ -1162,17 +1162,10 @@ impl<'a> ClarityDatabase<'a> {
         let value = Value::Tuple(
             TupleData::from_data(vec![
                 (
-                    ClarityName::try_from("reporter").map_err(|_| {
-                        InterpreterError::Expect("BUG: valid string representation".into())
-                    })?,
+                    ClarityName::from("reporter"),
```

Comment:
This is changing an error to a panic. That's probably okay in this case, since the input is a constant that we know will not error, but we should be careful with this kind of change. @kantai are you comfortable with leaving a panic in cases like this?

Reply:
There are several cases of this in this PR. I won't comment on the others, but if we decide not to make this change, all of the others should be reverted as well.

Reply:
Didn't realize there was a panic inside `ClarityName::from()`. If we're not even comfortable using this function on constant `&str`s, then what's the point of having it in the codebase? For test code only? In that case, we should make it explicit and add `#[cfg(test)]`

Reply:
I'm not opposed to that idea.

Reply:
Yeah -- these change lines should be reverted, because they're actually changing semantics here.

I agree we should make `ClarityName::from` test code only, but let's do that in a separate PR.

Reply:
Made an issue for this, #4398